web/bulk_rendering/renderer_core.ts

```diff
@@ -116,10 +116,14 @@ namespace com.keyman.renderer {
         let renderLayer = function(i: number) {
           return new Promise(function(resolve) {
             // (Private API) Directly sets the keyboard layer within KMW, then uses .show to force-display it.
-            if(isMobile) {
-              keyman.osk.layerId = layers[i].id;
+            if(keyman.osk.vkbd) {
+              if(isMobile) {
+                keyman.osk.vkbd.layerId = layers[i].id;
```

Comment:
Is line 109 up to date?
```
layers = keyman.osk.layers;
```

The page is working in desktop. But mobile viewport keeps flickering one OSK and the console log is filled with:
> Failed to load the "..." keyboard for rendering!

Finally
```
Uncaught TypeError: Cannot read property 'files' of undefined   kmnative.ts:341
```

Reply:
Good catch.